Include cl build instructions

dav-og opened the pull request:
working with Chrono v9.0.1 and Visual Studio 2022

dav-og committed:
Include cl build instructions

working with Chrono v9.0.1 and Visual Studio 2022

MShabara commented on README.md:
````diff
@@ -22,3 +22,57 @@ Visit the [HydroChrono website](https://nrel.github.io/HydroChrono/) for informa
+3. Configure and build the project:
+```powershell
+cmake .. -DChrono_DIR="<path_to_chrono_build>/cmake" -DHDF5_DIR="<path_to_hdf5_cmake>" -DPython3_ROOT_DIR="<path_to_python>"
+cmake --build . --config Release
````
In Line 52, we can add `-DCMAKE_BUILD_TYPE=Release ` just to make the user aware of this option and that it can be changed. 


Note: It is important that the build type used when configuring and building HydroChrono matches the build type of the Project Chrono libraries it depends on.
For example, if Project Chrono was built using RelWithDebInfo, HydroChrono should be built with the same configuration to avoid linker errors and ensure compatibility.

dav-og replied:
Thanks Mohamed!

This comes down to which platform you are building on - I've added [a new documentation file](
https://github.com/NREL/HydroChrono/pull/67/files#diff-e96e1c5d1c26327737668b7eb2258988bcc03cce46166dcc83db619d5e402d34) to provide additional information about CMake and how to use it on different platforms, including when to use 'CMAKE_BUILD_TYPE'.

I've also provided additional information in the readme to remind people to use the same build configuration in Chrono and HydroChrono (with **⚠️ Important:** to make it stand out) 👍

I've also included an example of how to use CMAKE_BUILD_TYPE for some basic Mac/Linux build instructions.

dav-og replied:
Let me know if anything else needs to be made clearer!
Thanks

MShabara replied:
This looks good to me,

I will go ahead and merge it to the main branch.

MShabara commented on README.md:
````diff
@@ -22,3 +22,57 @@ Visit the [HydroChrono website](https://nrel.github.io/HydroChrono/) for informa
+
+4. Run the tests:
+```powershell
+ctest -C Release --output-on-failure
````
Some tests failed on my side 
The following tests FAILED:
          2 - chloadaddedmass_01 (Exit code 0xc0000135)        core examples small
          3 - chrono_error_01 (Exit code 0xc0000135)           core examples small
         21 - demo_rm3_reg_waves_01 (Exit code 0xc0000409)     core demos long
         24 - demo_oswec_reg_waves_01 (Timeout)                 core demos long
Errors while running CTest


Its better to move the Post-Build steps to be before step 4 "Run the tests"

dav-og replied:
Those tests are being addressed in the other PR I've opened 👍

Have moved 'post build steps' to after 'run the tests' - that makes sense 👍

MShabara commented on docs/_main_pages/developer_docs/build_instructions.md:
```diff
@@ -8,48 +8,142 @@ parent_section: Developer Documentation
-   - Irrlicht.dll (note: no `.pdb` for Irrlicht)
-   - Other `ChronoEngine_moduleName.dll` and `ChronoEngine_moduleName.pdb` files as relevant.
+   - ChronoEngine_irrlicht.dll (if using Irrlicht)
+   - Irrlicht.dll (if using Irrlicht)
```
In the readme.md, `ChronoEngine.dll` is included

dav-og replied:
Its listed in both

MShabara reviewed (commented):
I tried building with HDF5 1.10.8 and it works fine with Visual Studio 2019, but it doesn't seem to be compatible with VS2022.
I also tested with the latest version of [HDF5](https://github.com/HDFGroup/hdf5/releases, ), but ran into several linker errors.

I was able to get things working by modifying the `target_link_libraries` section as follows:
```
target_link_libraries(HydroChrono
  PUBLIC
    ${CHRONO_LIBRARIES}
    ${HDF5_LIBRARIES}
    shlwapi
)
```

I also made sure to set the HDF5 flags:
```
set(HDF5_USE_STATIC_LIBRARIES OFF)
find_package(HDF5 REQUIRED COMPONENTS CXX)
```

dav-og commented:
cheers Mohamed

>> I was able to get things working by modifying the target_link_libraries section as follows:

Do you mean you were able to get it working with VS2022? Or is that still not working for you?

MShabara commented:
Hi Dave,

It worked for VS 2022 and the latest version of HDF5 but I had to modify the  `CMakeLists.txt`

dav-og committed:
Clarify build instructions for Windows and macOS/Linux, with platform-specific guidance

- Split build instructions into separate sections for Windows and macOS/Linux
- Explained differences in setting build type: --config (Windows) vs -DCMAKE_BUILD_TYPE (Unix)
- Emphasized the need to match HydroChrono’s build type with Project Chrono’s
- Moved post-build steps to occur before running tests
- Updated clean build instructions to reflect platform differences

dav-og committed:
Clarify build type consistency and platform scope in developer docs

- Added note on matching build type with Project Chrono
- Clarified Windows-specific scope of instructions

dav-og committed:
Add CMake build configuration guide

Adds a new markdown guide explaining single- vs. multi-config CMake generators, build types, and best practices for consistent builds across platforms.

dav-og commented:
I think 'shlwapi' is probably just required by the latest version of HDF5 - lets leave that from this PR and have a separate 'update HDF5 version' PR, which can cover the CMake changes and documentation/instructions/URL updates for working with the latest HDF5 version.

dav-og commented:
manually merged via GitHub Desktop commit d4766c3447f90ea0cd6dbb5d57728714729d2a21
